### PR TITLE
xrdp: use patched xrdp sources when building xorgxrdp

### LINKED ISSUE #249273: `xorgxrdp` uses unpatched `xrdp` sources
### Describe the bug
`xrdp` sources are patched with additional [fields](https://github.com/NixOS/nixpkgs/blob/b51db7ec1be773cc478a3f7060e04721f2a5c881/pkgs/applications/networking/remote/xrdp/dynamic_config.patch#L9) of xrdp client config. Unfortunately, `xorgxrdp` packaged with unpatched `xrdp` sources as seen [here](https://github.com/NixOS/nixpkgs/blob/b51db7ec1be773cc478a3f7060e04721f2a5c881/pkgs/applications/networking/remote/xrdp/default.nix#L31). This causes `model` and `layout` fields be overridden by value of `rsakeys_ini_file`.

### Steps To Reproduce
1. Enable `xrdp` service by
```nix
{ services.xrdp.enable = true; }
```
2. Login to xrdp session.
3. Check xkb settings.
```console
$ setxkbmap -print -verbose 10
Setting verbose level to 10
locale is C
Trying to load rules file ./rules/base...
Trying to load rules file /nix/store/01s51slwz2i6z019p6qb9298ws86zmq5-setxkbmap-1.3.2/share/X11/xkb/rules/base...
Success.
Applied rules from base:
rules:      base
model:      /run/xrdp/rsakeys.ini
layout:     s.ini
Trying to build keymap using the following components:
keycodes:   xfree86+aliases(qwerty)
types:      complete
compat:     complete
symbols:    pc+s.ini
geometry:   pc(pc104)
xkb_keymap {
	xkb_keycodes  { include "xfree86+aliases(qwerty)"	};
	xkb_types     { include "complete"	};
	xkb_compat    { include "complete"	};
	xkb_symbols   { include "pc+s.ini"	};
	xkb_geometry  { include "pc(pc104)"	};
};
```

### Expected behavior
xkb settings should contain appropriate values. For example: `model` - pc104 and and `layout` - us.

### Notify maintainers
@chvp 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.44, NixOS, 23.05 (Stoat), 23.05.20230809.9034b46`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.5`
 - channels(root): `""`
 - nixpkgs: `/home/adomas/.config/flakeInputs/nixpkgs`
```

### PR DESCRIPTION
**Pull request by chvp**
## Description of changes

Fixes #249273. @adomixaszvers can you verify that this also fixes the issue for you?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by adomixaszvers** [comment]
@chvp, it is working fine :).
```console
❯ setxkbmap -print -verbose 10
Setting verbose level to 10
locale is C
Trying to load rules file ./rules/base...
Trying to load rules file /nix/store/3l18zaf6mi7fby8a6amsh2nlldrcbax4-setxkbmap-1.3.2/share/X11/xkb/rules/base...
Success.
Applied rules from base:
rules:      base
model:      pc104
layout:     lt
Trying to build keymap using the following components:
keycodes:   xfree86+aliases(qwerty)
types:      complete
compat:     complete
symbols:    pc+lt
geometry:   pc(pc104)
xkb_keymap {
	xkb_keycodes  { include "xfree86+aliases(qwerty)"	};
	xkb_types     { include "complete"	};
	xkb_compat    { include "complete"	};
	xkb_symbols   { include "pc+lt"	};
	xkb_geometry  { include "pc(pc104)"	};
};
```
`lt` stands for Lithuanian layout which I use.

**Comment by Carunga** [comment]
> This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:
> 
> https://discourse.nixos.org/t/xrdp-and-non-english-keyboard-layout/32920/2

The patch solves my issue where I was not able to use a German keyboard via xrdp. Thank you, great work!

File: pkgs/applications/networking/remote/xrdp/default.nix [hunk 1]
```diff
@@ -28,29 +41,20 @@ let
 
     preConfigure = "./bootstrap";
 
-    configureFlags = [ "XRDP_CFLAGS=-I${xrdp.src}/common"  ];
+    configureFlags = [ "XRDP_CFLAGS=-I${patchedXrdpSrc}/common"  ];
```
Comment (alyssais) [inline pkgs/applications/networking/remote/xrdp/default.nix:44]: Would `srcOnly xrdp` work?  That way it wouldn't be necessary to pull out `patchedXrdpSrc` into a let binding.

Comment (chvp) [reply pkgs/applications/networking/remote/xrdp/default.nix:44]: It results in infinite recursion, sadly. (The `postInstall` of the `srcOnly` derivation still refers to `xorgxrdp`.)

**Comment by adomixaszvers** [comment]
Could this be backported to nixos-23.05?

**Comment by alyssais** [comment]
I've just noticed the hash changed — what's up with that?

**Comment by chvp** [comment]
Bad rebase after https://github.com/NixOS/nixpkgs/pull/252502. Sorry about that, my mistake. I've made a PR to fix this here: https://github.com/NixOS/nixpkgs/pull/255480

**Pull request merged**